internal/groups/groups.go

```diff
@@ -0,0 +1,112 @@
+package groups
+
+import (
+	"context"
+)
+
+type Member interface{}
+type Metadata map[string]interface{}
+
+type Group struct {
+	ID          string
+	OwnerID     string
+	ParentID    string
+	Name        string
+	Description string
+	Metadata    Metadata
+	// Level if retrieved with Parents or Children method
+	// indicates a level in hierarchy from first group node.
```

Comment:
Is level 0 when the Group is not fetched using Parents/Children? This can be confusing and misleading if not documented. Also, consider using a pointer (or at least setting it to a negative value) so that there is a difference between 0 level and level not fetched at all.